src/Security/Authentication/Certificate/ref/Microsoft.AspNetCore.Authentication.Certificate.netcoreapp.cs

```diff
@@ -24,6 +24,7 @@ public partial class CertificateAuthenticationOptions : Microsoft.AspNetCore.Aut
     {
         public CertificateAuthenticationOptions() { }
         public Microsoft.AspNetCore.Authentication.Certificate.CertificateTypes AllowedCertificateTypes { [System.Runtime.CompilerServices.CompilerGeneratedAttribute] get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute] set { } }
+        public System.Security.Cryptography.X509Certificates.X509Certificate2Collection CustomTrustStore { [System.Runtime.CompilerServices.CompilerGeneratedAttribute] get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute] set { } }
```

Comment:
In X509Chain we made it explicit that you had to set the TrustMode.  (Just adding something to the collection will fail, since you're in conflicting modes).

I recommend being explicit here, too.

Reply:
I always set TrustMode to `CustomRootTrust` when the trust store contains certificates - logic for this is [here](https://github.com/aspnet/AspNetCore/blob/1d9d78d4be5ec33191b46a9361fcc342a0a1cc7c/src/Security/Authentication/Certificate/src/CertificateAuthenticationHandler.cs#L170).

I'll expose the existing TrustMode property from CertificateAuthenticationOptions, unless you'd prefer that I use a separate enum and convert the values.

Reply:
The same enum is fine with me.  The concern is that if/when we add more modes this code can't adapt if it implicitly sets the mode by there being certs in this collection.

So please don't do anything implicit.